tests/test_transplants.py

```diff
@@ -1748,3 +1749,110 @@ def test_blocker_prevent_symlinks(phabdouble):
         )
         == "Revision introduces symlinks in the files `blahfile_symlink`."
     ), "Diff with symlinks present should fail the check."
+
+
+TRY_TASK_CONFIG_DIFF = """
+diff --git a/blah.json b/blah.json
+new file mode 100644
+index 0000000..663cbc2
+--- /dev/null
++++ b/blah.json
+@@ -0,0 +1 @@
++{"123":"456"}
+diff --git a/try_task_config.json b/try_task_config.json
+new file mode 100644
+index 0000000..e44d36d
+--- /dev/null
++++ b/try_task_config.json
+@@ -0,0 +1 @@
++{"env": {"TRY_SELECTOR": "fuzzy"}, "version": 1, "tasks": ["source-test-cram-tryselect"]}
+""".lstrip()
+
+
+def test_blocker_try_task_config_no_landing_state(phabdouble):
+    repo = phabdouble.repo()
+
+    revision = phabdouble.revision(repo=repo)
+    phab_revision = phabdouble.api_object_for(
+        revision,
+        attachments={"reviewers": True, "reviewers-extra": True, "projects": True},
+    )
+    diff = phabdouble.diff(revision=revision, rawdiff=TRY_TASK_CONFIG_DIFF)
+
+    phab_client = phabdouble.get_phabricator_client()
+    stack_data = request_extended_revision_data(phab_client, [revision["phid"]])
+
+    # Parse diffs into `rs_parsepatch` format.
+    parsed_diffs = get_parsed_diffs(phab_client, stack_data)
+    stack_state = create_state(stack_data=stack_data, parsed_diffs=parsed_diffs)
```

Comment:
I'm a little concerned that we are circumventing the creation of stack state through the actual implementation and just using the `create_state` method that is defined in this test module (see my broader comment on the PR). It looks like the only way a `StackAssessmentState` is ever created is via the `from_assessment` class method, so why not try and use that for these tests as well? 

If there are optional arguments that aren't needed, even for testing, then they should be optional in `from_assessment`. But if they aren't optional, they can be included via other fixtures (e.g., `relman_project`, `mocked_repo_config`, etc...)

I think ideally this would all be done in `create_state` similar to how you have it, but though the use of fixtures, and by avoiding adding any random values if they aren't actually used for testing. There are fixtures for almost all if not all the things required by `StackAssessmentState.from_assessment`, and those that do not have fixtures are things that can be created using fixtures.

Here's a quick example of what that could look like.

```python
@pytest.fixture
def create_state(phabdouble, mocked_repo_config, ...):
    def _create_state(revision_phid, *kwargs):
        phab = phabdouble.get_phabricator_client()
        stack_data = request_extended_revision_data(phab, [revision["phid"]])
        supported_repos = get_repos_for_env("test")

        # ... leverage other available fixtures here to fetch all required parameters
        return StackAssessmentState.from_assessment(...)
    return _create_state
```

then

```suggestion
    stack_state = create_state(revision["phid"], ...)
```

In summary, these tests exist withing the ecosystem of the other tests, which have various dependencies that are provided by existing fixtures.

I should have caught this problem in the previous PR, but this PR made it more obvious that these tests are not leveraging existing fixtures as much as they should be.

Reply:
These tests are just supposed to be unit tests for the checks, so we only really need to populate the fields required to test the check function logic. This is what was done to test the warnings/blockers in the pre-refactor state. For example, previously even though the function `warning_previously_landed` took all the kwargs passed to all the warnings, we only populate the `revision` and `diffs` arguments in `test_warning_previously_landed_*` test functions, since those are the only arguments required to run the warning.

Having a more robust test fixture is a good idea and I'll look into adding it as a follow-up, but given it isn't regressing anything to keep the tests as-is, I don't think it should block this PR.

Reply:
Filed [bug 1896501](https://bugzilla.mozilla.org/show_bug.cgi?id=1896501) for this and will have a few PRs up shortly, once those land we can move forward with this review.